src/java.base/share/classes/sun/security/pkcs/SigningCertificateInfo.java

```diff
@@ -81,10 +81,16 @@
  */
 public class SigningCertificateInfo {
 
+    private byte[] ber;
     private ESSCertId[] certId = null;
 
     public SigningCertificateInfo(byte[] ber) throws IOException {
         parse(ber);
+        this.ber = ber;
+    }
+
+    public byte[] toByteArray() {
```

Comment:
Hm, maybe making this class package private as well?  The returned array is not cloned.   It may not worthy of a clone as it is used just once in the package.  But if it is public, a lot of time would be costed in the future to check if is a potential issue.